runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,353 @@
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+type Client struct {
+	dockerClient *client.Client
+	config       *Config
+}
+
+func New(_ context.Context) (*Client, error) {
+	config, err := NewConfig()
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration. Provider=%s: %w", models.Docker, err)
+	}
+
+	if err = config.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate provider configuration. Provider=%s: %w", models.Docker, err)
+	}
+
+	dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, fmt.Errorf("failed to load provider configuration. Provider=%s: %w", models.Docker, err)
+	}
+
+	return &Client{
+		dockerClient: dockerClient,
+		config:       config,
+	}, nil
+}
+
+func (c *Client) Kind() models.CloudProvider {
+	return models.Docker
+}
+
+func (c *Client) DiscoverAssets(ctx context.Context) ([]models.AssetType, error) {
+	// Get image assets
+	imageAssets, err := c.getImages(ctx)
+	if err != nil {
+		return nil, provider.FatalError{
+			Err: fmt.Errorf("failed to get images. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Get container assets
+	containerAssets, err := c.getContainers(ctx)
+	if err != nil {
+		return nil, provider.FatalError{
+			Err: fmt.Errorf("failed to get containers. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Combine assets
+	assets := append(imageAssets, containerAssets...)
+
+	return assets, nil
+}
+
+func (c *Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+
+	err := c.prepareScanVolume(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to prepare scan volume. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.createScanConfigFile(config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scanconfig.yaml file. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	containerId, err := c.createScanContainer(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.dockerClient.ContainerStart(ctx, containerId, types.ContainerStartOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to start scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) prepareScanVolume(ctx context.Context, config *provider.ScanJobConfig) error {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return err
+	}
+
+	// Create volume if not found
+	resp, err := c.dockerClient.VolumeList(ctx, volume.ListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get volumes. Provider=%s: %w", models.Docker, err)
+	}
+	if len(resp.Volumes) == 1 {
+		fmt.Printf("scan volume already created. Provider=%s", models.Docker)
```

Comment:
Avoid `fmt` package to print logging messages. Also, note that `fmt` package is not thread-safe.
You can create a logger from context via `logger := log.GetLoggerFromContextOrDiscard(ctx)` and use that.